dataplane/event/eventor.go

```diff
@@ -117,14 +123,24 @@ func (e *Eventor) ProcessSample(samplerLogs dsample.SamplerOTLPLogs) error {
 	// it's call, new events will not be visited.
 	dsample.RangeSamplerLogsWithType[dsample.RawSampleOTLPLog](samplerLogs, func(rawSample dsample.RawSampleOTLPLog) {
 		for _, event := range e.events {
-			if slices.Contains(rawSample.Streams(), event.streamUID) {
+			if slices.Contains(rawSample.Streams(), event.stream.UID) {
 				data, err := rawSample.SampleData()
 				if err != nil {
 					errs = errors.Join(errs, err)
 					continue
 				}
 
-				ruleMatches, err := event.rule.Eval(context.Background(), data)
+				var ruleMatches bool
+				if event.stream.Keyed != nil {
+					ruleMatches, err = event.rule.EvalKeyed(context.Background(), rawSample.SampleKey(), data)
+					// In case of reaching the maximum number of keys, we continue to the next event without
+					// logging any error
+					if err != nil && err == function.ErrMaxKeys {
```

Comment:
it should be fine to return the error, these errors are received by the Processor and logged using a rate limited logger. May be a bit noisy but for now, may be better to notify somehow instead of silently producing incomplete results

Reply:
Yeah, I did not know what to do in that case, but now that you mention the rate limiting, I think I will log the error.